hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/SimpleRegionNormalizer.java

```diff
@@ -382,15 +398,16 @@ private List<NormalizationPlan> computeMergeNormalizationPlans(final NormalizeCo
           break;
         }
         if (
-          rangeMembers.isEmpty() // when there are no range members, seed the range with whatever
-                                 // we have. this way we're prepared in case the next region is
-                                 // 0-size.
+          (rangeMembers.isEmpty() // when there are no range members, seed the range with whatever
+                                  // we have. this way we're prepared in case the next region is
+                                  // 0-size.
             || (rangeMembers.size() == 1 && sumRangeMembersSizeMb == 0) // when there is only one
                                                                         // region and the size is 0,
                                                                         // seed the range with
                                                                         // whatever we have.
             || regionSizeMb == 0 // always add an empty region to the current range.
-            || (regionSizeMb + sumRangeMembersSizeMb <= avgRegionSizeMb)
+            || (regionSizeMb + sumRangeMembersSizeMb <= avgRegionSizeMb))
+            && (rangeMembers.size() < getMergeMaxRegionCount())
```

Comment:
Please invert the logic to be another `|| condition` -- adding the `&&` makes the logic more confusing.